src/cata_tiles.cpp

```diff
@@ -179,66 +193,212 @@ void cata_tiles::get_tile_information(std::string config_path, std::string &json
     }
 }
 
+inline pixel cata_tiles::get_pixel_color(SDL_Surface *surf, int x, int y, int w)
```

Comment:
These two pixel adaptors (translating from a SDL pixel to your idea of a pixel IIUC) don't need to be part of the cata_tiles class, you can make them static helper functions, which will limit their level of privledge and not expose them to callers of cata_tiles.